contracts/simple-ica-controller/src/ibc.rs

```diff
@@ -111,11 +117,60 @@ pub fn ibc_packet_ack(
 #[allow(clippy::unnecessary_wraps)]
 fn acknowledge_dispatch(
     _deps: DepsMut,
+    _env: Env,
     _caller: String,
-    _ack: StdAck,
+    callback: Option<String>,
+    msg: IbcPacketAckMsg,
 ) -> Result<IbcBasicResponse, ContractError> {
     // TODO: actually handle success/error?
-    Ok(IbcBasicResponse::new().add_attribute("action", "acknowledge_dispatch"))
+    match callback {
```

Comment:
This is a great addition.

I am skeptical of allowing the caller to set who is called back (rather than just set true/false on callback, leading to a callback to the given contract). I will merge, but make a PR on top to clean up this logic